frontend/javascripts/oxalis/model/accessors/volumetracing_accessor.ts

```diff
@@ -416,6 +416,34 @@ export function getMappingInfoForVolumeTracing(
   return getMappingInfo(state.temporaryConfiguration.activeMappingByLayer, tracingId);
 }
 
+export function hasEditableMapping(state: OxalisState): boolean {
+  const volumeTracing = getActiveSegmentationTracing(state);
+
+  if (volumeTracing == null) return false;
+
+  return !!volumeTracing.mappingIsEditable;
+}
+
+export function isMappingActivationAllowed(
+  state: OxalisState,
+  mappingName: string | null | undefined,
+): boolean {
+  const isEditableMappingActive = hasEditableMapping(state);
+
+  if (isEditableMappingActive) {
+    const volumeTracing = getActiveSegmentationTracing(state);
+
+    // This should never be the case, since editable mappings can only be active for volume tracings
+    if (volumeTracing == null) return false;
+
+    // Only allow mapping activations of the editable mapping itself if an editable mapping is saved
+    // in the volume tracing. Editable mappings cannot be disabled or switched for now.
+    return mappingName === volumeTracing.mappingName;
```

Comment:
I agree with the early-out :+1:
However, I'm not sure about the `mappingName == null`. What exactly would the condition look like?
In my view, the current condition is testing the right thing. If `mappingName` is null, but `volumeTracing.mappingName` is not null, the mapping (de)activation is not allowed (correct). If `mappingName` is null, and `volumeTracing.mappingName` also is null (far fetched, because `mappingIsEditable` should not be true then), the mapping (de)activation is allowed (also correct imo).